src/main/scala/cromiam/samiam/SamIamPerCreateRegistrationActor.scala

```diff
@@ -0,0 +1,57 @@
+package cromiam.samiam
+
+import akka.actor.{Actor, ActorLogging, ActorRef, Props, Terminated}
+import cromiam.samiam.SamIamActor.SamFailed
+import cromiam.samiam.SamIamPerCreateRegistrationActor.SendMe
+import cromiam.samiam.SamSupervisor.SamActorDown
+
+import scala.concurrent.Promise
+
+class SamIamPerCreateRegistrationActor(user: String,
+                                       workflowId: String,
+                                       samIamActor: ActorRef,
+                                       completeMe: Promise[Unit]) extends Actor with ActorLogging {
+
+  context.watch(samIamActor)
+  log.info(s"Registering new workflow creation '$workflowId' by user '$user'")
+  self ! SendMe(SamIamActor.RegisterWorkflow(user, workflowId))
+
+  override def receive: Receive = {
+    case SendMe(msg) => samIamActor ! msg
+    case registrationComplete @ SamIamActor.WorkflowRegistrationComplete(u, w) =>
+      if (user == u && workflowId == w) {
+        complete(())
+      } else {
+        complete(new Exception(s"Authorization response didn't correlate to request: SamIamActor.RegisterWorkflow($user, $workflowId) returned $registrationComplete"))
+      }
+    case SamFailed =>
+      complete(SamActorDown)
+    case Terminated(`samIamActor`) =>
+      complete(complete(SamActorDown))
+    case other =>
+      complete(new Exception(s"Unexpected response from SamIamActor: $other"))
+  }
+
+  private def complete(u: Unit) = {
```

Comment:
Yeah, this is probably a leftover from when it was actually returning a real thing so it should probably disappear